### PR TITLE
feat: Add optional security context for eventbus container template

### PR DESCRIPTION
**Pull request by hamza3202**
Adds option to specify container security context for event-bus
#1509

### CODE REVIEW

File: pkg/apis/eventbus/v1alpha1/types.go [hunk 1]
```diff
@@ -145,6 +145,7 @@ type NativeStrategy struct {
 type ContainerTemplate struct {
 	Resources       corev1.ResourceRequirements `json:"resources,omitempty" protobuf:"bytes,1,opt,name=resources"`
 	ImagePullPolicy corev1.PullPolicy           `json:"imagePullPolicy,omitempty" protobuf:"bytes,2,opt,name=imagePullPolicy,casttype=PullPolicy"`
+	SecurityContext corev1.SecurityContext      `json:"securityContext,omitempty" protobuf:"bytes,3,opt,name=securityContext"`
```
Comment (whynowy) [inline pkg/apis/eventbus/v1alpha1/types.go:148]: Use pointer please.

Comment (whynowy) [reply pkg/apis/eventbus/v1alpha1/types.go:148]: Also add the logic in controller code https://github.com/argoproj/argo-events/blob/master/controllers/eventbus/installer/nats.go#L708-L751

**Comment by whynowy** [comment]
Please git commit with `-s`

File: pkg/apis/eventbus/v1alpha1/types.go (hunk 1 above)
Comment (hamza3202) [reply pkg/apis/eventbus/v1alpha1/types.go]: Done.

**Comment by hamza3202** [comment]
@whynowy There were some un-related changes when I ran `make codegen` that I had to revert manually in `api/*.md` files. Not sure if they should be pushed or not? The changes were of two kinds mainly.
- Empty line after each line
- Escaping certain characters e.g `#` in URLs and some conditional signs e.g `!,<` etc

**Review by whynowy — APPROVED** [review]
LGMT, thanks!

**Comment by hamza3202** [comment]
@whynowy when can i expect this to be released?

**Pull request merged**